atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/store/AtlasDbTransactionStoreTest.java

```diff
@@ -162,6 +166,29 @@ public void abortedTransactionsReturnEmpty() {
                 .isEmpty();
     }
 
+    @Test
+    public void keyAlreadyExistExceptionResultsInMaybeWitnessedTransaction() {
+        WriteTransactionAction writeTransactionAction =
+                WriteTransactionAction.of(TABLE_1, WORKLOAD_CELL_ONE, VALUE_ONE);
+        TransactionManager onlyThrowsKeyAlreadyExistsExceptionManager = spy(manager);
+        doAnswer(answer -> {
+                    Supplier<CommitTimestampProvider> commitTimestampFetcher = answer.getArgument(0);
+                    ConditionAwareTransactionTask<Void, CommitTimestampProvider, Exception> task =
+                            answer.getArgument(1);
+                    manager.runTaskWithConditionWithRetry(commitTimestampFetcher, task);
+                    throw new KeyAlreadyExistsException("foobar");
+                })
+                .when(onlyThrowsKeyAlreadyExistsExceptionManager)
+                .runTaskWithConditionWithRetry(any(Supplier.class), any());
+        AtlasDbTransactionStore onlyKeyAlreadyExistsThrowingStore =
+                AtlasDbTransactionStore.create(onlyThrowsKeyAlreadyExistsExceptionManager, TABLES_TO_ATLAS_METADATA);
+        assertThat(onlyKeyAlreadyExistsThrowingStore.readWrite(List.of(writeTransactionAction)))
+                .hasValueSatisfying(witnessedTransaction -> {
+                    assertThat(witnessedTransaction).isInstanceOf(MaybeWitnessedTransaction.class);
+                    assertThat(witnessedTransaction.actions()).containsExactly(writeTransactionAction.witness());
```

Comment:
re-reading this, probably should have checked that commit timestamp is present, but this is a precondition when we construct `MaybeWitnessedTransaction`, thus is not needed